syntax = "proto3";

package mycqu_service;

import "micro_services_protobuf/mycqu_service/mycqu_model.proto";
import "micro_services_protobuf/mycqu_service/mycqu_request_response.proto";

service MycquFetcher {
    rpc FetchUser(BaseLoginInfo) returns (UserInfo);    // 获取用户信息

    rpc FetchEnrollCourseInfo(FetchEnrollCourseInfoRequest) returns (FetchEnrollCourseInfoResponse);    // 获取可选课程信息
    rpc FetchEnrollCourseItem(FetchEnrollCourseItemRequest) returns (FetchEnrollCourseItemResponse);    // 获取可选具体课程信息

    rpc FetchExam(FetchExamRequest) returns (FetchExamResponse);    // 获取考试信息

    rpc FetchAllSession(BaseLoginInfo) returns (FetchAllSessionResponse);   // 获取全部学期信息
    rpc FetchCurrSessionInfo(BaseLoginInfo) returns (CquSessionInfo);   // 获取当前学期起止日信息
    rpc FetchAllSessionInfo(BaseLoginInfo) returns (FetchAllSessionInfoResponse);   // 获取全部学期起止日信息
    rpc FetchCourseTimetable(FetchCourseTimetableRequest) returns (FetchCourseTimetableResponse);   // 获取课表
    rpc FetchEnrollTimetable(FetchEnrollTimetableRequest) returns (FetchCourseTimetableResponse);   // 获取选课信息

    rpc FetchScore(FetchScoreRequest) returns (FetchScoreResponse); // 获取成绩信息
    rpc FetchGpaRanking(BaseLoginInfo) returns (GpaRanking); // 获取绩点排名信息
}

service CardFetcher {
    rpc FetchCard(BaseLoginInfo) returns (Card);    // 获取校园卡余额
    rpc FetchBills(BaseLoginInfo) returns (FetchBillResponse);  // 获取最近30天账单信息
    rpc FetchEnergyFee(FetchEnergyFeeRequest) returns (EnergyFees); // 获取水电费信息
}
